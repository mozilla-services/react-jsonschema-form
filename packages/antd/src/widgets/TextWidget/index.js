import React from 'react';

import Input from 'antd/lib/input';
import InputNumber from 'antd/lib/input-number';
import { rangeSpec } from '@rjsf/core/lib/utils'

const INPUT_STYLE = {
  width: '100%',
};

const TextWidget = ({
  // autofocus,
  disabled,
  formContext,
  id,
  // label,
  onBlur,
  onChange,
  onFocus,
  options,
  placeholder,
  readonly,
  // required,
  schema,
  value,
}) => {
  const { readonlyAsDisabled = true } = formContext;

  const handleNumberChange = (nextValue) => onChange(nextValue);

  const handleTextChange = ({ target }) =>
    onChange(target.value === '' ? options.emptyValue : target.value);

  const handleBlur = ({ target }) => onBlur(id, target.value);

  const handleFocus = ({ target }) => onFocus(id, target.value);

  const step = schema.type === 'number' ? 'any' : undefined // non-integer numbers shouldn't have a default step of 1
  const stepProps = rangeSpec(schema) // sets step, min, and max from the schema

  return schema.type === 'number' || schema.type === 'integer' ? (
    <InputNumber
      disabled={disabled || (readonlyAsDisabled && readonly)}
      id={id}
      name={id}
      onBlur={!readonly ? handleBlur : undefined}
      onChange={!readonly ? handleNumberChange : undefined}
      onFocus={!readonly ? handleFocus : undefined}
      placeholder={placeholder}
      style={INPUT_STYLE}
      type="number"
      step={step}
      {...stepProps}
      value={value}
    />
  ) : (
    <Input
      disabled={disabled || (readonlyAsDisabled && readonly)}
      id={id}
      name={id}
      onBlur={!readonly ? handleBlur : undefined}
      onChange={!readonly ? handleTextChange : undefined}
      onFocus={!readonly ? handleFocus : undefined}
      placeholder={placeholder}
      style={INPUT_STYLE}
      type={options.inputType || 'text'}
      value={value}
    />
  );
};

export default TextWidget;
