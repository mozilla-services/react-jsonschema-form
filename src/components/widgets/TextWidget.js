import React, { PropTypes } from "react";


function TextWidget({
  schema,
  id,
  placeholder,
  value,
  required,
  onChange
}) {
  return (
    <input type="text"
      id={id}
      className="form-control"
      value={typeof value === "undefined" ? "" : value}
      placeholder={placeholder}
      required={required}
      onChange={(event) => onChange(event.target.value)} />
  );
}

if (process.env.NODE_ENV !== "production") {
  TextWidget.propTypes = {
    schema: PropTypes.object.isRequired,
    id: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    value: PropTypes.oneOfType([
      React.PropTypes.string,
      React.PropTypes.number,
    ]),
    required: PropTypes.bool,
    onChange: PropTypes.func,
  };
}

export default TextWidget;
