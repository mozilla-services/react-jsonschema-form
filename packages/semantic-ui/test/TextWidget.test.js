import React from "react";
import TextWidget from "../src/TextWidget";
import renderer from "react-test-renderer";

const noop = () => undefined;
describe("TextWidget", () => {
  const props = {
    uiSchema: {},
    onChange: noop,
    onBlur: noop,
    onFocus: noop,
    label: 'test',
    disabled: false,
    rawErrors: [""],
    value: "value",
    options: {},
    formContext: {},
    id: "id",
  }
  test("schema integer type as string", () => {
    const tree = renderer
      .create(<TextWidget {...props} schema={{type: 'integer'}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
