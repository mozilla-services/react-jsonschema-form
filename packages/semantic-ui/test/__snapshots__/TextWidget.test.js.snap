// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWidget schema integer type as string 1`] = `
<div
  className="field"
>
  <label
    htmlFor="id"
  >
    test
  </label>
  <div
    className="ui input"
  >
    <input
      id="id"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value="value"
    />
  </div>
</div>
`;
